src/mscorlib/shared/System/Guid.Unix.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    partial struct Guid
+    {
+        static Random _random = new Random();
```

Comment:
I do not think that `Random` is good enough for GUID generation. It needs to go to PAL Random, like this: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/Guid.Unix.cs